doc/source/rllib/rllib-offline.rst

```diff
@@ -8,217 +8,662 @@ Working With Offline Data
-- ``v_target``: The OPE's estimated discounted return for the target policy, averaged over episodes in the batch.
-- ``v_target_std``: The standard deviation corresponding to v_target.
-- ``v_gain``: ``v_target / max(v_behavior, 1e-8)``. ``v_gain > 1.0`` indicates that the policy is better than the policy that generated the behavior data. In case, ``v_behavior <= 0``, ``v_delta`` should be used instead for comparison.
-- ``v_delta``: The difference between v_target and v_behavior.
+.. code-block:: python
 
-As an example, we generate an evaluation dataset for off-policy estimation:
+    from ray.rllib.algorithms.bc import BCConfig
+    from ray.rllib.core.columns import Columns
 
-.. code-block:: bash
+    config = (
+        BCConfig()
+        ...
+        .offline_data(
+            input_=[<input_path>],
+            # Provide the schema of your data.
+            input_read_schema={
+                Columns.OBS: "o_t",
+                Columns.ACTIONS: "a_t",
+                Columns.REWARDS: "r_t",
+                Columns.NEXT_OBS: "o_tp1",
+                Columns.INFOS: "i_t",
+                "done": "d_t",
+            },
+        )
+    )
+        
+.. note:: Internally, the legacy ``gym``'s ``done`` signals are mapped to ``gymnasium``'s ``terminated`` signals, with ``truncated`` values defaulting to 
+    ``False``. RLlib's ``SingleAgentEpisode`` structures align with ``gymnasium``, adhering to the updated environment API standards in reinforcement learning.
 
-    $ rllib train \
-        --run=PG \
-        --env=CartPole-v1 \
-        --config='{"output": "/tmp/cartpole-eval", "output_max_file_size": 5000000}' \
-        --stop='{"timesteps_total": 10000}'
+Converting tabular data to RLlib's episode format
+-------------------------------------------------
+While the tabular format is widely compatible and seamlessly integrates with RLlib's new Offline RL API, there are cases where you may prefer to use RLlib's native episode format. 
+As briefly mentioned earlier, such scenarios typically arise when full expert trajectories are required.
 
-.. hint:: You should use separate datasets for algorithm training and OPE, as shown here.
+.. note::
+    RLlib processes tabular data in batches, converting each row into a *single-step episode*. This approach is primarily for procedural simplicity, as data cannot 
+    generally be assumed to arrive in time-ordered rows grouped by episodes, though this may occasionally be the case (however knowledge of such a structure resides  
+    with the user as RLlib cannot easily infer it automatically). While it is possible to concatenate ``SingleAgentEpisode`` chunks, this cannot be done with fragmented chunks.
 
-We can now train a DQN algorithm offline and evaluate it using OPE:
+If you require full trajectories you can transform your tabular data into ``SingleAgentEpisode`` objects and store these in Parquet format. The next example shows how to do this. 
```

Comment:
In general across this rst, always use the:
```
:py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode`
```

notation for classes and the:

```
:py:meth:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode.get_observations`
```

notation for individual class methods.

This automatically creates links into these APIs' reference pages.

Reply:
So replace all ``SingleAgentEpisode`` with
```
:py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode`
```

etc..

Reply:
@sven1977 Some of these classes and their methods appear not to be in the docs.